examples/nine-pt-stencil-solver/nine-pt-stencil-solver.cpp

```diff
@@ -0,0 +1,320 @@
+University of Tennessee
+
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its contributors
+   may be used to endorse or promote products derived from this software
+   without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+/*****************************<COMPILATION>***********************************
+The easiest way to build the example solver is to use the script provided:
+./build.sh <PATH_TO_GINKGO_BUILD_DIR>
+
+Ginkgo should be compiled with `-DBUILD_REFERENCE=on` option.
+
+Alternatively, you can setup the configuration manually:
+
+Go to the <PATH_TO_GINKGO_BUILD_DIR> directory and copy the shared
+libraries located in the following subdirectories:
+
+    + core/
+    + core/device_hooks/
+    + reference/
+    + omp/
+    + cuda/
+
+to this directory.
+
+Then compile the file with the following command line:
+
+c++ -std=c++11 -o nine-pt-stencil-solver nine-pt-stencil-solver.cpp -I../.. \
+    -L. -lginkgo -lginkgo_reference -lginkgo_omp -lginkgo_cuda
+
+(if ginkgo was built in debug mode, append 'd' to every library name)
+
+Now you should be able to run the program using:
+
+env LD_LIBRARY_PATH=.:${LD_LIBRARY_PATH} ./nine-pt-stencil-solver
+
+*****************************<COMPILATION>**********************************/
+
+#include <array>
+#include <chrono>
+#include <ginkgo/ginkgo.hpp>
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+// Comment or uncomment to choose a stencil.
+
+/*std::array<double,9> coefs{
+    -1.0/6.0, -2.0/3.0, -1.0/6.0,
+    -2.0/3.0, 10.0/3.0, -2.0/3.0,
+    -1.0/6.0, -2.0/3.0, -1.0/6.0};
+*/
+std::array<double, 9> coefs{-1.0, -1.0, -1.0, -1.0, 8.0,
```

Comment:
Regarding formatting, you can wrap it in comments like `// clang-format on` and `// clang-format off` to preserve the number of rows and columns so that the stencil can be read easily. See [here](https://github.com/ginkgo-project/ginkgo/blob/develop/reference/test/matrix/coo_kernels.cpp#L152) for an example. 

If you could move this to the main function and provide the user to choose this at runtime, that would be better as compiling the code to change the stencil is not good practice.

Reply:
I have done this.

Reply:
and I will rewrite that main function part.

Reply:
I have rewritten it.